include/ppx/application.h

```diff
@@ -559,6 +564,18 @@ class Application
     // Initializes standard knobs
     void InitStandardKnobs();
 
+    // List gpus
+    void ListGPUs() const;
+
+    // Returns true to exit the loop
+    bool Mainloop();
```

Comment:
The app already had a `IsRunning()` and `Quit()` method. Seems like this could return void, and the whole logic handled using those (see suggestions).